sql/driver.ts

```diff
@@ -0,0 +1,111 @@
+/** @fileoverview Package driver defines interfaces to be implemented by database drivers as used by package `x/database/sql`.
+
+ Inspired by https://golang.org/src/database/sql/driver/driver.go.
+*/
+
+// deno-lint-ignore-file no-empty-interface
+
+/** Type metadata associated with a driver. */
+export interface BaseMeta {
```

Comment:
There should be separate types for input and output I think, since some database drivers might accept different input and output types (e.g. SQLite supports `Date` as input, but internally stores them as strings, so the driver has no way to know if a string is a date, and thus will never return any `Date` objects back out)

Reply:
Ah, good to know, thanks! I thought `deno-sqlite` had the same input and output types, and I wasn't sure about Postgres, so I was going to use a single type for now, but if even `deno-sqlite` has different types I'll break them up from the start.

Reply:
Split into `BoundType` and `ResultType`.